src/app/profile/my-spaces/my-spaces.component.ts

```diff
@@ -55,28 +92,46 @@ export class MySpacesComponent implements OnDestroy, OnInit {
     private modalService: BsModalService,
     private spaceService: SpaceService,
     private userService: UserService,
-    private authentication: AuthenticationService
-  ) {
-    this.subscriptions.push(this.contexts.current.subscribe(val => this.context = val));
-    this.subscriptions.push(userService.loggedInUser.subscribe(user => {
-      this.loggedInUser = user;
+    private auth: AuthenticationService,
+    private http: HttpClient,
+    private errorHandler: ErrorHandler,
+    @Inject(WIT_API_URL) private readonly witUrl: string
+  ) { }
+
+  ngOnInit() {
+    this.subscriptions.push(
+      this.contexts.current.subscribe(val => this.context = val)
+    );
+    this.subscriptions.push(
+      this.userService.loggedInUser.subscribe(user => { this.loggedInUser = user; })
+    );
+    this.subscriptions.push(
+      this.broadcaster.on('displayMySpaces').subscribe((): void => {
```

Comment:
This works, but I wonder if using `Broadcaster` for this is sort of an abuse. It seems like this should be implemented with normal a standard Angular EventEmitter between the two components, rather than an application-wide message broadcast. No other components in the application should really care about this message.

Reply:
Yeah that sounds okay, I'll look into switching over to an EventEmitter instead of broadcasting.